### ISSUE TITLE
How to implement source only training?

### ISSUE THREAD

**Issue by hanyikao** [giuliomattolin/ConfMix#9]
Thank you for your great work.

In the README, the contents are about having training target domain data on hand. But currently I am working on a project that it is restricted to use any labels from target domain. I found that there are some dataset.yaml files which ends with "PseudoLabelling", however I have trouble with the directory of "FoggyFromCityscapes/images". What should I put inside this? I would greatful if you can help solving this issue.

**Comment by luca-zanella-dvl** [comment]
Hi Ololkao, thank you for your interest in our work. 

The files located in the `data/` directory with the suffix `PseudoLabelling.yaml` were used to generate the results for the ablation study titled "Does sample mixing perform better than simple self-training?" in the paper. In particular, these files are employed to fine-tune a model that has been trained on the source domain using a dataset that combines the original data from the source domain with the pseudo detections generated by the model itself on the target domain.

To generate the `FoggyFromCityscapes/images` folder, you need to follow these steps:

1. Download the source-only model `cityscapes.pt` from the drive folder specified in the readme and save it under the root directory (alternatively, you can train your own source-only model).
2. Once you have the cityscapes.pt file, you can use the following script:

```
python detect.py \
    --weights 'cityscapes.pt' \
    --source '/usr/src/datasets/FoggyCityscapes/images/train/*/*.png' \
    --imgsz 600 \
    --conf-thres 0.25 \
    --iou-thres 0.5 \
    --save-txt \
    --nosave \
    --project '/usr/src/datasets' \
    --name 'FoggyFromCityscapes'

mkdir /usr/src/datasets/FoggyFromCityscapes/images/

for f in $(find /usr/src/datasets/FoggyFromCityscapes/labels/ -type f -name '*.txt'); 
do
    basename=${f##*/}; 
    basename=${basename%.*}; 
    ln -s $(find /usr/src/datasets/FoggyCityscapes/images/train/ -name "${basename}.png") "/usr/src/datasets/FoggyFromCityscapes/images/"; 
done
```

After running this script, it creates the `/usr/src/datasets/FoggyFromCityscapes/images/` folder. Then, it iterates through the text files in `/usr/src/datasets/FoggyFromCityscapes/labels/` and creates symbolic links to the corresponding images in `/usr/src/datasets/FoggyCityscapes/images/train/`.
